### PR TITLE
Add `--strict-bytes` flag

### LINKED ISSUE #18256: Add --strict-bytes flag to disable promotions to bytes
**Feature**

We currently have the undocumented flags `--disable-bytearray-promotion` and `--disable-memoryview-promotion` that turn off promotions from `bytearray`/`memoryview` to `bytes`. These are a bit unwieldy, and I'm proposing that we add a new flag called `--strict-bytes` that enables both of these flags, and make it an officially documented feature. In mypy 2.0 we'd switch `--strict-bytes` on by default.

Additional details:
 * This should be documented.
 * Also support `--no-strict-bytes` for the inverse. This allows users to prepare for mypy 2.0 even if they aren't ready to turn on the flag.
 * This probably shouldn't be a per-module option, since migration to the new behavior appears to be fairly easy even for large codebases.
 * We'll still keep the old undocumented flags for now, but they might be removed in the future.

**Pitch**

Currently `bytearray` and `memoryview` are assignable to `bytes`, which is incorrect, but this has been tricky to fix mostly due to stub quality. Stubs seem to have improved enough for this to be practical now -- any anyway switching the default would give more incentives to fix stubs.

The new flag is shorter and easier to remember, and it combines both of the features so it's simpler. The naming is also arguably consistent with `--strict-equality` and friends. In my experience dropping the `memoryview` promotion rarely causes trouble, so having it as a separate flag doesn't seem important.

### PR DESCRIPTION
**Pull request by hamdanal**
Closes #18256

### CODE REVIEW

**Comment by sterliakov** [comment]
Could you (perhaps temporarily) enable this flag during primer runs? To quickly change that, there's `--additional-flags` param given to `mypy_primer` in `.github/workflows/mypy_primer.yml`. It'd be interesting to see the impact of this option on the open-source corpus.

**Comment by hamdanal** [comment]
> Could you (perhaps temporarily) enable this flag during primer runs? To quickly change that, there's `--additional-flags` param given to `mypy_primer` in `.github/workflows/mypy_primer.yml`. It'd be interesting to see the impact of this option on the open-source corpus.

See the primer hits here #12661

File: docs/source/command_line.rst [hunk 1]
```diff
@@ -659,6 +659,34 @@ of the above sections.
 
        assert text is not None  # OK, check against None is allowed as a special case.
 
+
+.. option:: --strict-bytes
```
Comment (JukkaL) [inline docs/source/command_line.rst:663]: Can you also add a short section about `strict_bytes` to `config_file.rst` as well?

**Review by JukkaL — COMMENTED** [review]
Thanks for the PR! Looks good, just one comment about documentation.

File: docs/source/command_line.rst (hunk 1 above)
Comment (hamdanal) [reply docs/source/command_line.rst:663]: Done

**Review by JukkaL — APPROVED** [review]

**Pull request merged**